stake-pool/single-pool/tests/helpers/token.rs

```diff
@@ -0,0 +1,54 @@
+#![allow(dead_code)]
+
+use {
+    mpl_token_metadata::{pda::find_metadata_account, state::Metadata},
+    solana_program_test::BanksClient,
+    solana_sdk::{
+        borsh::try_from_slice_unchecked,
+        hash::Hash,
+        message::Message,
+        program_pack::Pack,
+        pubkey::Pubkey,
+        signature::{Keypair, Signer},
+        transaction::Transaction,
+    },
+    spl_associated_token_account as atoken,
+    spl_token::state::{Account, Mint},
+};
+
+pub async fn create_ata(
```

Comment:
Not needed for this PR, but `TokenClient` was designed to avoid files like this, so whatever we need to eventually get rid of this and use `TokenClient` would be nice

Reply:
this is my first time thinking about using it outside the cli, and thinking as a user, the main ergonomic hurdle is constructing the `ProgramClient`. i want a function like `build_from_mint_address` or something that i just pass in... i guess my rpc or banks client and the mint address and it makes the program client, fetches the mint, fills out the decimals and program id. just a first idea

Reply:
That makes sense. We should consider adding `ProgramClient` into one of the monorepo crates so that the pattern is easier to follow for other program clients, along with some nice helpers like you're suggesting.